libhb/nvenc_common.c

```diff
@@ -73,12 +137,44 @@ int hb_nvenc_h264_available()
 
 int hb_nvenc_h265_available()
 {
-    return hb_check_nvenc_available();
+    if (is_nvenc_hevc_available != -1)
+    {
+        return is_nvenc_hevc_available;
+    }
+    
+    if (!hb_check_nvenc_available()){
+        is_nvenc_hevc_available = 0;
+        return is_nvenc_hevc_available;
+    }
+    
+    if (hb_nvenc_get_cuda_version() >= 6.0) {
+        is_nvenc_hevc_available = 1;
+    }else {
+        is_nvenc_hevc_available = 0;
+    }
+        
+    return is_nvenc_hevc_available;
 }
 
 int hb_nvenc_av1_available()
 {
-    return hb_check_nvenc_available();
```

Comment:
I wonder whether similiar checks could be added for hb_nvdec_available. E.g. AV1 decode is only available since Ampere.

Reply:
The current method for decode appears to already work.  AV1 decode on my system with a 3070 is using software, not hardware.  It appears to be probing via ffmpeg functions.

Reply:
> The current method for decode appears to already work. AV1 decode on my system with a 3070 is using software, not hardware. It appears to be probing via ffmpeg functions.

That should be right. We offering FFmpeg a CUDA HW context that it can decide to use. But your 3070 has support for AV1 decode, it should use AV1 HW decode were possible https://developer.nvidia.com/video-encode-and-decode-gpu-support-matrix-new